ncm-opennebula/src/main/perl/OpenNebula/Account.pm

```diff
@@ -0,0 +1,188 @@
+    my %newaccounts = map { $_ => 1 } @datalist;
+
+    foreach my $t (@existsdata) {
+        # Remove the user/group only if the QUATTOR flag is set
+        my $quattor = $self->check_quattor_tag($t,1);
+        if (exists($protected{$t->name})) {
+            $self->info("This $type is protected and can not be removed: ", $t->name);
+        } elsif (exists($newaccounts{$t->name})) {
+            $self->verbose("$type required by Quattor. We can't remove it: ", $t->name);
+        } elsif (!$quattor) {
+            $self->verbose("QUATTOR flag not found. We can't remove this $type: ", $t->name);
+        } else {
+            push(@rmdata, $t->name);
+            $t->delete();
+        }
+    }
+
+    if (@rmdata) {
+        $self->info("Removed ${type}s: ", join(',', @rmdata));
+    }
+
+    foreach my $account (sort keys %{$data}) {
+        if (exists($protected{$account})) {
+            $self->info("This $type is protected and can not be created/updated: ", $account);
+        } else {
+            $temp{$account}->{$account} = $data->{$account};
+            $template = $self->process_template($temp{$account}, $type);
+            my $used = $self->detect_used_resource($one, $type, $account);
+            if (!$used) {
+                $self->info("Creating new $type: ", $account);
+                if (exists $data->{$account}->{password}) {
+                    # Create new user
+                    $one->$createmethod($account, $data->{$account}->{password}, $CORE_AUTH_DRIVER);
+                    $self->change_user_group($one, $account, $data->{$account}->{group}) if (exists $data->{$account}->{group});
+                } else {
+                    # Create new group
+                    $one->$createmethod($account);
+                };
+                # Add Quattor flag
+                $new = $self->update_something($one, $type, $account, $template);
+            } elsif ($used == -1) {
+                # User/group is already there and we can modify it
+                $self->info("Updating $type with a new template: ", $account);
+                $new = $self->update_something($one, $type, $account, $template);
+                if ((exists $data->{$account}->{group}) and ($type eq "user")) {
+                    $self->change_user_group($one, $account, $data->{$account}->{group});
+                };
+            }
+        }
+    }
+}
+
+=item change_user_group
+
+Sets user primary group.
+
+=cut
+
+sub change_user_group
+{
+    my ($self, $one, $name, $group) = @_;
+
+    $self->info("User: $name must belong to this primary group: ", $group);
+    my $group_id = $self->get_resource_id($one, "group", $group);
+
+    if (defined($group_id)) {
+        my @users = $one->get_users(qr{^$name$});
+        $self->warn("Detected administration group change for regular user: $name") if ($group_id == 0);
+        foreach my $usr (@users) {
+            my $out = $usr->chgrp($group_id);
+            if (defined($out)) {
```

Comment:
since you do not do anything with `$out`, you can also do `if (defined($usr->chgrp($group_id)))`